### PR TITLE
Puppeteer

### PR DESCRIPTION
**Pull request by meirish**
Hello!

I was thinking about using this for an app I work on but I had a few incompatibilities. 

1. the app I work on uses `rootURL` in its config, and this didn't seem to be supported
2. I wasn't super keen on having to export the app globally to get this working

After poking around a bit I figured out how to get `rootURL` working with the express server, and it seemed like moving to puppeteer directly might help with the other problem. Sadly there doesn't seem to be a way to make critical work with the HTML + CSS that's output when using a `rootURL`, so I added a `skipCritical` option that just re-writes the HTML file without going through critical.

So TLDR; this is working for me and seems to simplify the code generally - I thought I'd see if there was interest in having this upstreamed. If not, that's ok too!

Let me know what you think, and thanks for making this 👍.

### CODE REVIEW

**Comment by meirish** [comment]
Gonna close for now this doesn't seem quite ready - waitForSelector doesn't seem to behave properly... I'll open a new one once I figure that out.
